modules/python/src/custom_nodes/google_genmedia/utils.py

```diff
@@ -166,89 +172,26 @@ def generate_image_from_text(
         safety_filter_level=safety_filter_level,
     )
 
-    retries = 0
     generated_pil_images: List[PIL_Image.Image] = []
-    while retries <= retry_count:
-        try:
-            print("Sending request to Imagen API for text-to-image generation...")
-            response = client.models.generate_images(
-                model=model, prompt=prompt, config=config
-            )
+    print("Sending request to Imagen API for text-to-image generation...")
+    response = client.models.generate_images(model=model, prompt=prompt, config=config)
```

Comment:
this should be in a try except block to catch the cascading exception and then print it and exit the program.